### PR TITLE
Implement PriorityCommandParser class and PriorityCommand class

### PR DESCRIPTION
**Pull request by PresleyChew**
fixes #73

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 1]
```diff
@@ -14,6 +14,7 @@ public class CliSyntax {
     public static final Prefix PREFIX_DOB = new Prefix("d/");
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
     public static final Prefix PREFIX_TAG = new Prefix("t/");
+    public static final Prefix PREFIX_PRIORITY = new Prefix("!/");
```
Comment (TechRaj) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:17]: might be a bit risky to use exclamation mark. Maybe use p safer?

File: src/main/java/seedu/address/model/person/Priority.java [hunk 2]
```diff
@@ -12,18 +12,18 @@
 public class Priority {
 
     // Regex to validate the priority value (LOW, MEDIUM, or HIGH).
-    public static final String VALIDATION_REGEX = "LOW|MEDIUM|HIGH";
+    public static final String VALIDATION_REGEX = "NONE|LOW|MEDIUM|HIGH";
```
Comment (TechRaj) [inline src/main/java/seedu/address/model/person/Priority.java:15]: smart

**Review by TechRaj — COMMENTED** [review]
LGTM just some nitpicks

File: src/main/java/seedu/address/model/person/Priority.java [hunk 3]
```diff
@@ -12,18 +12,18 @@
...
 
     // Error message for invalid priority values.
-    public static final String MESSAGE_CONSTRAINTS = "Priority should only be 'LOW', 'MEDIUM', or 'HIGH'.";
+    public static final String MESSAGE_CONSTRAINTS = "Priority should only be 'NONE', 'LOW', 'MEDIUM', or 'HIGH'.";
 
     private final String priority;
```
Comment (yadobler) [inline src/main/java/seedu/address/model/person/Priority.java:20]: Actually I wonder if using Enum would have been easier, then to get the string use `Enum.valueOf(x)` and stuff. But for now it looks good to me.

File: src/test/java/seedu/address/logic/parser/PriorityCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    @Test
+    public void parse_compulsoryFieldsMissing_failure() {
+        // Both parameters missing
+        assertParseFailure(parser, "", String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_USAGE));
+
+        // Nric parameter missing
+        assertParseFailure(parser, PRIORITY_DESC_AMY, String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_USAGE));
+
+        // Priority parameter missing
+        assertParseFailure(parser, NRIC_DESC_AMY, String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_USAGE));
```
Comment (yong-zhuo) [inline src/test/java/seedu/address/logic/parser/PriorityCommandParserTest.java:81]: Test cases are comprehensive 👍🏻

File: src/main/java/seedu/address/logic/parser/PriorityCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        System.out.println(trimmedArgs);
+        if (!argMultimap.getValue(PREFIX_NRIC).isPresent()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    PriorityCommand.MESSAGE_USAGE));
+        }
+
+        if (!argMultimap.getValue(PREFIX_PRIORITY).isPresent()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    PriorityCommand.MESSAGE_USAGE));
+        }
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/logic/parser/PriorityCommandParser.java:42]: Edge cases are properly handled

**Review by yong-zhuo — APPROVED** [review]
LGTM 🔥

**Pull request merged**